Conceptpower+Spring/src/main/java/edu/asu/conceptpower/rest/ConceptIDLookup.java

```diff
@@ -150,4 +150,58 @@
 
     }
 
+    /**
+     *
+     * This method provides original concept information for the rest interface
+     * of the form
+     * "http://[server.url]/conceptpower/rest/Concept?id={IDs of concept seperated by ,}"
+     * 
+     * @param ids
+     *            - Ids based on which concepts will be fetched.
+     * @param acceptHeader
+     *            - Decides the type of response. Can be application/xml or
+     *            application/json. Default value is application/xml.
+     * @return
+     * @throws JsonProcessingException
+     */
+    @RequestMapping(value = "/OriginalConcepts", method = RequestMethod.GET, produces = {
+            MediaType.APPLICATION_XML_VALUE, MediaType.APPLICATION_JSON_VALUE })
+    public @ResponseBody ResponseEntity<String> getOriginalConceptById(@RequestParam String ids,
+            @RequestHeader(value = "Accept", defaultValue = MediaType.APPLICATION_XML_VALUE) String acceptHeader)
+                    throws JsonProcessingException {
+
+        if (ids == null || ids.trim().isEmpty()) {
+            return new ResponseEntity<String>(HttpStatus.BAD_REQUEST);
+        }
+
+        String[] wordnetIds = ids.split(",");
+
+        if (wordnetIds == null) {
+            return new ResponseEntity<String>(HttpStatus.BAD_REQUEST);
+        }
+
+        Set<ConceptEntry> conceptEntries = new HashSet<>();
+        for (String wordnetId : wordnetIds) {
+            if (!wordnetId.trim().equalsIgnoreCase("")) {
+                ConceptEntry entry = conceptManager.getOriginalConceptEntry(wordnetId);
+                conceptEntries.add(entry);
+            }
+        }
+
+        IConceptMessage msg = messageFactory.getMessageFactory(acceptHeader).createConceptMessage();
+        Map<ConceptEntry, ConceptType> entryMap = new LinkedHashMap<>();
+        if (conceptEntries != null && !conceptEntries.isEmpty()) {
```

Comment:
I don't think you need the isEmpty check here. it should just skip the loop if the set is empty I believe. And since you initialized the set further up, it can never be null.